### PR TITLE
Map Refactor

### PR DESCRIPTION
**Pull request by Donwilo**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
Map functionality relies on class-based components.
Key operations such as deleting, editing, creating polygons, and zooming are handled by class methods.
Map control functions are tightly coupled with class-based implementation, making the code less modular and harder to maintain.

Issue Number: N/A

## What is the new behavior?
Refactored the map functionality to use hooks and functional components.
Delete Polygon: Now streamlined for better performance and readability using hooks.
Edit Polygon: Enhanced with hooks for more intuitive state management.
Create Polygon: Rewritten as a functional component for consistency and ease of use.
Zoom In/Out: Improved control and responsiveness using hooks.
Map Control Functions: Refactored to be more modular and maintainable.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

This refactor introduces breaking changes due to the shift from class-based components to functional components with hooks.
Existing applications using the old class-based methods will need to be updated to the new hook-based approach.

## Other information
We did not finish the TypeScript typing validation. Jorge and Cesar, please help us with this part to ensure all types are correctly validated.

### CODE REVIEW

File: src/admin/components/ResourceTabs/PolygonReviewTab/index.tsx [hunk 1]
```diff
@@ -98,6 +103,27 @@ const PolygonReviewTab: FC<IProps> = props => {
 
   const [polygonFromMap, setPolygonFromMap] = useState<IpolygonFromMap>({ isOpen: false, uuid: "" });
 
+  async function storePolygon(geojson: any, record: any) {
+    if (geojson?.length) {
```
Comment (cesarLima1) [inline src/admin/components/ResourceTabs/PolygonReviewTab/index.tsx:107]: is there a way to avoid 'any'?

**Comment by cesarLima1** [comment]
for future integration of endpoints, we use `generate:api` and the definitions in `swagger_v2` that generates automatically the apiComponent and apiSchemas

**Review by cesarLima1 — APPROVED** [review]

File: src/admin/components/ResourceTabs/PolygonReviewTab/index.tsx [hunk 2]
```diff
@@ -98,6 +103,27 @@ const PolygonReviewTab: FC<IProps> = props => {
 
   const [polygonFromMap, setPolygonFromMap] = useState<IpolygonFromMap>({ isOpen: false, uuid: "" });
 
+  async function storePolygon(geojson: any, record: any) {
+    if (geojson && geojson[0]) {
```
[the remark this answers is not in the document]
Comment (Donwilo) [reply src/admin/components/ResourceTabs/PolygonReviewTab/index.tsx]: done

File: src/components/elements/Map-mapbox/utils.ts [hunk 3]
```diff
@@ -1,3 +1,263 @@
...
+
+export const loadLayersInMap = (map: mapboxgl.Map, polygonsData: SitePolygonsDataResponse | undefined) => {
+  layersList.forEach((layer: any) => {
+    if (map) {
+      showPolygons(layer.styles, layer.name, map, "uuid", polygonsData);
+    }
+  });
+};
+
+function handleLayerClick(
```
[the remark this answers is not in the document]
Comment (Donwilo) [reply src/components/elements/Map-mapbox/utils.ts]: done

File: src/components/elements/Map-mapbox/utils.ts [hunk 4]
```diff
@@ -1,3 +1,263 @@
...
+
+    targetLayers.forEach(targetLayer => {
+      map.on("click", targetLayer.id, (e: any) =>
+        handleLayerClick(e, popupComponent, map, setPolygonFromMap, sitePolygonData)
+      );
+    });
+  }
+};
+
+export function addSourceToLayer(layer: any, map: mapboxgl.Map, polygonsData: SitePolygonsDataResponse | undefined) {
```
[the remark this answers is not in the document]
Comment (Donwilo) [reply src/components/elements/Map-mapbox/utils.ts]: done

File: src/components/elements/Map-mapbox/utils.ts [hunk 5]
```diff
@@ -1,3 +1,263 @@
...
+    type: "vector",
+    tiles: [URL_GEOSERVER]
+  });
+  styles?.forEach((style: LayerWithStyle, index: number) => {
+    addLayerStyle(map, name, style, index);
+  });
+  loadLayersInMap(map, polygonsData);
+}
+
+export function addLayerStyle(map: mapboxgl.Map, sourceName: string, style: LayerWithStyle, index: number) {
```
[the remark this answers is not in the document]
Comment (Donwilo) [reply src/components/elements/Map-mapbox/utils.ts]: done

File: src/components/elements/Map-mapbox/utils.ts [hunk 6]
```diff
@@ -1,3 +1,263 @@
...
+import { BBox, FeatureCollection } from "./GeoJSON";
+import type { LayerType, LayerWithStyle } from "./Map.d";
+
+const GEOSERVER = "https://geoserver-prod.wri-restoration-marketplace-api.com";
+
 export function getFeatureProperties<T extends any>(properties: any, key: string): T | undefined {
   return properties[key] || properties[`user_${key}`];
 }
+
+export function convertToGeoJSON(featureCollection: FeatureCollection) {
```
[the remark this answers is not in the document]
Comment (Donwilo) [reply src/components/elements/Map-mapbox/utils.ts]: done

**Pull request merged**